newsroom/topics/views.py

```diff
@@ -87,75 +107,66 @@ async def update_topic(topic_id):
     for subscriber in updates["subscribers"]:
         subscriber["user_id"] = ObjectId(subscriber["user_id"])
 
-    if (
-        original
-        and updates.get("is_global") != original.get("is_global")
-        and original.get("folder") == updates.get("folder")
-    ):
+    if original and updates.get("is_global") != original.is_global and original.folder == updates.get("folder"):
         # reset folder when going from company to user and vice versa
         updates["folder"] = None
 
-    response = get_resource_service("topics").patch(id=ObjectId(topic_id), updates=updates)
+    await TopicService().update(args.topic_id, updates)
+
+    topic = await TopicService().find_by_id(args.topic_id)
 
-    auto_enable_user_emails(updates, original, current_user)
+    await auto_enable_user_emails(updates, original, current_user)
 
-    if response.get("is_global") or updates.get("is_global", False) != original.get("is_global", False):
+    if topic.is_global or updates.get("is_global", False) != original.is_global:
         push_company_notification("topics")
     else:
         push_user_notification("topics")
-    return jsonify({"success": True}), 200
+
+    return Response({"success": True})
 
 
-@blueprint.route("/topics/<topic_id>", methods=["DELETE"])
+@topic_endpoints.endpoint("/topics/<string:topic_id>", methods=["DELETE"])
 @login_required
-async def delete(topic_id):
+async def delete(args: RouteArguments, params: None, request: Request):
     """Deletes a followed topic by given id"""
+    service = TopicService()
     current_user = get_user(required=True)
-    original = get_resource_service("topics").find_one(req=None, _id=ObjectId(topic_id))
+    original = await service.find_by_id(args.topic_id)
 
-    if not can_edit_topic(original, current_user):
+    if not await can_edit_topic(original, current_user):
         abort(403)
```

Comment:
we should use the abort method from the `request` view param to avoid having much dependencies on quart/flask
```suggestion
        await request.abort(403)
```